bin/oref0-pushover.sh

```diff
@@ -28,3 +28,94 @@ elif [[ $ONLYFOR =~ "insulin" ]] && ! cat $FILE | egrep "maxBolus"; then
 else
     curl -s -F "token=$TOKEN" -F "user=$USER" -F "sound=$SOUND" -F "message=$(jq -c "{bg, tick, carbsReq, insulinReq, reason}|del(.[] | nulls)" $FILE) - $(hostname)" https://api.pushover.net/1/messages.json && touch monitor/pushover-sent
 fi
+
+# Send to pushover glances API 
+#   For watch face updates once every 10 minutes
+#   Works for apple watch complications 
+#   Not tested fo android wear notifications yet
+
+# Use file to make sure we don't update glances more often than every 10 mins
+# Apple watch disables any complication updates that happen more frequently
+
+source $HOME/.bash_profile
+key=${MAKER_KEY:-"null"}
+carbsReq=`jq .carbsReq ${FILE}`
+tick=`jq .tick ${FILE}`
+bgNow=`jq .bg ${FILE}`
+delta=`echo "${tick}" | tr -d +`
+delta="${delta%\"}"
+delta="${delta#\"}"
+cob=`jq .COB $FILE`
+iob=`jq .IOB $FILE`
+
+#echo "carbsReq=${carbsReq} tick=${tick} bgNow=${bgNow} delta=${delta} cob=${cob} iob=${iob}"
+pushoverGlances=$(cat preferences.json | jq -M '.pushoverGlances')
+
+if [ "${pushoverGlances}" == "null" -o "${pushoverGlances}" == "false" ]; then
+    echo "No preference or false preference for pushoverGlances"
+else
+  GLANCES="monitor/last_glance"
+  GLUCOSE="monitor/glucose.json"
+  if [ ! -f $GLANCES ]; then
+    # First time through it will get created older than 10 minutes so it'll fire
+    touch $GLANCES && touch -r $GLANCES -d '-11 mins' $GLANCES
+  fi
+
+  if test `find $GLANCES -mmin +10`
+  then
+    enactTime=$(ls -l  --time-style=+"%l:%M" ${FILE} | awk '{printf ($6)}')
+
+   direction=""
+    if [ "${delta}" -lt "-7" ]; then
+      direction="--"
+    elif [ "${delta}" -lt "-2" ]; then
+      direction="-"
+    elif [ "${delta}" -lt "3" ]; then
+      direction=""
+    elif [ "${delta}" -lt "8" ]; then
+      direction="+"
+    else
+      direction="++"
+    fi
```

Comment:
Normally direction arrows are based on something like a 15m average.  Do we have the avgDelta available to use here?

Reply:
Is there an avgDelta stored somewhere in an openaps json file? I could calculate it from glucose.json if not.

Reply:
No, we calculate avgDelta in determine-basal, but don't write it out anywhere.  I guess we could either add that to the enacted json, or recalculate it here.  Or you could do something more like what @cjo20 did with the mdt-trend-arrow stuff...

Reply:
I looked at the mdt-trend-arrow code and it definitely looks like something that xdrip-js-logger should be doing. If it is done there, then any downstream consumer of xdrip-js-logger glucose data would have better trend arrows including NS. Let me look at that in the next couple of days. Then this apple watch pushover glances update could just use the trend arrow from the glucose data directly. @scottleibrand - What do you think?